spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/SubscribingByGroupOperation.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import java.util.function.Consumer;
+
+/**
+ * Operations for subscribing to a destination with a consumer group.
+ *
+ * @author Warren Zhu
+ */
+public interface SubscribingByGroupOperation<D> {
+
+    /**
+     * Register a message consumer to a given destination with a given consumer group.
+     *
+     * @return {@code true} if the consumer was subscribed or {@code false} if it
+     * was already subscribed.
+     */
+    boolean subscribe(String destination, Consumer<Iterable<D>> consumer, String consumerGroup);
```

Comment:
I'd rather make interface method simplist possible. e.g. avoid generics if can.

Reply:
Generic has to be used here to improve re usability. Reasons as before.